core/src/main/java/org/elasticsearch/action/update/UpdateRequestBuilder.java

```diff
@@ -279,23 +330,23 @@ public UpdateRequestBuilder setUpsert(Object... source) {
         return this;
     }
 
-    public UpdateRequestBuilder setSource(XContentBuilder source) throws Exception {
-        request.source(source);
+    public UpdateRequestBuilder fromXContent(XContentBuilder source) throws Exception {
```

Comment:
I wonder if we can just move the source parsing into the rest handler or into a static method on the request - do we really need it as part of the java API?